lib/ddg-atb.js

```diff
@@ -35,16 +36,52 @@ function redirect(e) {
   };
 }
 
+function update(){
+    var atb     = ss.storage.atb, 
+        setATB  = ss.storage.set_atb;
+
+    if(!atb || !setATB){
+        return;
+    }
+
+    var xhr = XMLHttpRequest();
+
+    xhr.onreadystatechange = function() {
+        var DONE = XMLHttpRequest.DONE ? XMLHttpRequest.DONE : 4;
+        if(xhr.readyState == DONE){
+            if(xhr.status == 200) {
+                var curATB = JSON.parse(xhr.responseText);
+                if(curATB.version !== setATB){
+                    ss.storage.set_atb = curATB.version
+                }
+            }
+        }
+    };
+
+    xhr.open('GET',
+            'https://duckduckgo.com/atb.js?' + Math.ceil(Math.random() * 1e7) 
+            + '&atb=' + atb + '&set_atb=' + setATB,
+            true
+    );
+
+    xhr.send();
+}
 
 exports.install = function () {
-  WebRequest.onBeforeSendHeaders.addListener(redirect,
-                                             {
-                                               urls: pattern,
-                                               types: ["main_frame", "sub_frame"]
-                                             },
-                                             ["blocking"]);
+  WebRequest.onBeforeSendHeaders.addListener(redirect, 
+          {
+              urls: pattern,
+              types: ["main_frame", "sub_frame"]
+          },
+          ["blocking"]);
+  
+  WebRequest.onCompleted.addListener(update, 
+          { 
+              urls: pattern
```

Comment:
Do we need the `main_frame`/`sub_frame` types like in the redirect listener? I don't really know what they do.